File: go-sdk/kestra_api_client/kill_switches_api.go
package kestra_api_client

import "context"

// KillSwitchesAPI covers the instance-level /api/v1/kill-switches routes. Kill
// switches pause execution evaluation across the instance; every endpoint
// requires the caller to be an instance owner and is not tenant-scoped.
type KillSwitchesAPI struct {
	baseAPI
}

// SearchKillSwitches returns every kill switch across all tenants.
func (a *KillSwitchesAPI) SearchKillSwitches(ctx context.Context) ([]KillSwitch, error) {
	return doJSON[[]KillSwitch](&a.baseAPI, ctx, "GET", superadminPath("kill-switches", "search"), nil, nil)
}

// CreateKillSwitch creates a kill switch and broadcasts it to the cluster.
func (a *KillSwitchesAPI) CreateKillSwitch(ctx context.Context, killSwitch KillSwitch) (*KillSwitch, error) {
	result, err := doJSON[KillSwitch](&a.baseAPI, ctx, "POST", superadminPath("kill-switches"), killSwitch, nil)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// UpdateKillSwitch updates the kill switch with the given id. The body id must
// match the path id or the server answers 422.
func (a *KillSwitchesAPI) UpdateKillSwitch(ctx context.Context, id string, killSwitch KillSwitch) (*KillSwitch, error) {
	result, err := doJSON[KillSwitch](&a.baseAPI, ctx, "PUT", superadminPath("kill-switches", id), killSwitch, nil)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// DeleteKillSwitch deletes the kill switch with the given id.
func (a *KillSwitchesAPI) DeleteKillSwitch(ctx context.Context, id string) error {
	return a.doVoid(ctx, "DELETE", superadminPath("kill-switches", id), nil, nil)
}

File: python/python-sdk/kestrapy/api/kill_switches_api.py
from typing import List

from kestrapy.base_api import BaseApi
from kestrapy.models.kill_switch import KillSwitch


class KillSwitchesApi(BaseApi):
    """Kill switches (`/api/v1/kill-switches`). Instance-owner-only."""

    def create_kill_switch(self, kill_switch: KillSwitch) -> KillSwitch:
        path = self._superadmin_path("kill-switches")
        return self._json_request("POST", path, KillSwitch, body=kill_switch)

    def update_kill_switch(self, id: str, kill_switch: KillSwitch) -> KillSwitch:
        path = self._superadmin_path("kill-switches", id)
        return self._json_request("PUT", path, KillSwitch, body=kill_switch)

    def delete_kill_switch(self, id: str) -> None:
        path = self._superadmin_path("kill-switches", id)
        self._void_request("DELETE", path)

    def search_kill_switches(self) -> List[KillSwitch]:
        path = self._superadmin_path("kill-switches", "search")
        return self._json_list_request("GET", path, KillSwitch)
